packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -57,91 +55,66 @@ const Root = ({
-
-  const showControls = isInteracting || (showControlsWhenPaused && !isPlaying)
+  const showControlsRow1 = showPlayPauseButton || showSeekControl
+  const showControlsRow2 = showCCButton || showMuteButton || showVolumeControl || showFullScreenButton
+  const showControlsBar = showControlsRow1 || showControlsRow2
 
   return (
-    <div className={styles.VideoPlayer__container} ref={videoWrapperRef}>
-      <video ref={ref} title={title} controls={false} className={clsx(styles.VideoPlayer, className)} {...rest}>
-        {children}
-        <track kind="captions" />
-      </video>
-      {showBranding || !visuallyHiddenTitle ? (
-        <div className={clsx(styles.VideoPlayer__title, isPlaying && styles['VideoPlayer__title--hidden'])}>
-          {showBranding && <MarkGithubIcon size={40} />}
-          {!visuallyHiddenTitle && (
-            <Text size="400" weight="medium" className={styles.VideoPlayer__controlTextColor}>
-              {title}
-            </Text>
+    <div className={styles.VideoPlayer__container} ref={fullscreenRef}>
+      <div className={styles.VideoPlayer__overlayContainer}>
+        <video ref={ref} title={title} controls={false} className={clsx(styles.VideoPlayer, className)} {...rest}>
+          {children}
+          <track kind="captions" />
+        </video>
+        {showBranding || !visuallyHiddenTitle ? (
+          <div className={clsx(styles.VideoPlayer__title, isPlaying && styles['VideoPlayer__title--hidden'])}>
+            {showBranding && <MarkGithubIcon size={40} />}
+            {!visuallyHiddenTitle && (
+              <Text size="400" weight="medium" className={styles.VideoPlayer__controlTextColor}>
+                {title}
+              </Text>
+            )}
+          </div>
+        ) : null}
+        <button
+          className={clsx(
+            styles.VideoPlayer__playButtonOverlay,
+            isPlaying && styles['VideoPlayer__playButtonOverlay--transparent'],
           )}
-        </div>
-      ) : null}
-      <button
-        className={clsx(
-          styles.VideoPlayer__playButtonOverlay,
-          isPlaying && styles['VideoPlayer__playButtonOverlay--transparent'],
-        )}
-        onClick={togglePlaying}
-        aria-label={isPlaying ? 'Pause' : 'Play'}
-      >
-        {!isPlaying && <PlayIcon />}
-      </button>
-      <div className={styles.VideoPlayer__controls}>
-        {ccEnabled && <Captions />}
-        <ControlsBar className={clsx(!showControls && styles['VideoPlayer__controlsBar--fade'])}>
-          {showPlayPauseButton && <PlayPauseButton />}
-          {showSeekControl && <SeekControl />}
-          {showCCButton && <CCButton />}
-          {showMuteButton && <MuteButton />}
-          {showVolumeControl && !isSmall && <VolumeControl />}
-          {showFullScreenButton && <FullScreenButton />}
-        </ControlsBar>
+          onClick={togglePlaying}
+          aria-label={isPlaying ? 'Pause' : 'Play'}
+        >
+          {!isPlaying && <PlayIcon />}
```

Comment:
Should this be ternary instead with a pause icon (or null) as the fallback?

Reply:
No, this is the overlay that's used on paused videos. Adding a ternary with a pause icon here would result in a pause button being overlayed on top of playing videos.